pkg/cmd/roachprod-microbench/compare.go

```diff
@@ -66,15 +82,19 @@ func newCompare(config compareConfig) (*compare, error) {
 		return nil, err
 	}
 	ctx := context.Background()
-	service, err := google.New(ctx)
-	if err != nil {
-		return nil, err
+	var service *google.Service
+	if config.threshold == skipComparison {
```

Comment:
In that sense, do we want to create a new flag `--post-to-slack` too or do we want to club that logic in something like `--publish-results` flag?

Reply:
The current logic already gates slack based on if a slack token is sent. Google sheets do not need a token as it uses the Google Creds available on the environment. So I think we'd only need to have an exclusive flag to prevent publishing google sheets. I think there's still value in separating slack posting & google sheet generation, since someone might run a job to investigate the sheets, but don't want it to post to slack (which is mostly reserved for the weekly job).